### ISSUE TITLE
Using install_rpath for AIX

### ISSUE THREAD

**Issue by KamathForAIX** [mesonbuild/meson#13981]
Hi community members,

Currently in AIX, there is no way [as far as I have searched] to pass install_path during the build while using Meson unless and until a project has given provision for the same using an option.

What we are looking for is something like [CMAKE_INSTALL_RPATH](https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_RPATH.html).

Is there any way we can do the same in Meson?

If we cannot, then in AIX, we do pass in an environment variable called LD_FLAGS. 

For example,
export LDFLAGS="-Wl,-blibpath:/opt/freeware/lib/pthread:/opt/freeware/lib64:/opt/freeware/lib:/usr/lib:/lib"

When we pass like this gcc understands  that this is a linker flag and we are asking the compiler to set the LIBPATH within the XCOFF object to the blibpath mentioned in LDFLAGS.

By doing so when we dump a shared object we get,
![image](https://github.com/user-attachments/assets/c32ed6c6-e81c-4f80-ab8d-b85d5270aa91)

If you see the path is what I have given from LDFLAGS when I created this library manually. [See Import File strings Index 0]

When we compile a package through pip in particular there is no way to tell meson that Hey we are looking to add the LDFLAGS blibpath that should override the build_rpath.

Hence I would like to add this feature in Meson for AIX by making some changes [here](https://github.com/mesonbuild/meson/blob/master/mesonbuild/linkers/linkers.py#L1523).

Kindly let me know if we can proceed to add this via LDFLAGS environment variable or if the community has some other way to achieve the same.

**Comment by eli-schwartz** [comment]
On Linux, the meson.build can set an rpath to be used and the GNU option `-Wl,-rpath=...` can be passed both via LDFLAGS and via e.g. pkg-config dependencies.

Meson will also pass a few of its own during the build in case of libraries and executables that depend on each other, so that running the software directly in `./builddir/...` will work. At install time, it will rewrite the binary file if it can, using mesonbuild/scripts/depfixer.py, to remove build-time-only paths and add meson.build `install_rpath` entries.

Are the correct rpaths present when you inspect the build directory? Do they just disappear during install? If they only disappear during install then it's possible the depfixer.py routines are broken on AIX. We have some logic in there to figure out which RPATH entries come from LDFLAGS and preserve them, but it's possible this only got wired up for GNU.

...

It is absolutely expected behavior that if you add this via LDFLAGS it should be respected.

**Comment by KamathForAIX** [comment]
@eli-schwartz 

>Meson will also pass a few of its own during the build in case of libraries and executables that depend on each other, so that running the software directly in ./builddir/... will work.

I get this. That is why we are able to run the meson test.

>Are the correct rpaths present when you inspect the build directory?
Yes.

>Do they just disappear during install?
No. But the issue is that the paths passed via -Wl,blibpath in the LDFLAGS environment variable are not considered by meson as of now in AIX.


>We have some logic in there to figure out which RPATH entries come from LDFLAGS and preserve them, but it's possible this only got wired up for GNU.

Yes, I see that the logic is for opening an ELF format binary in depfixer.py and changing the libpath. But this will not work for the XCOFF object file in AIX. But I think we should do the same for AIX. 

Can we develop something similar for AIX in depfixer.py with a different class, may be XCOFF? The requirement is that if we have blibpath from LDFLAGS environment variables, then we simply add that along with the required rpaths at install time.

**Comment by eli-schwartz** [comment]
Yes please :) do whatever it takes to make sure that meson's AIX handling supports the same features that the Linux handling does.
